sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/ShareFileStorageResourceOptions.cs

```diff
@@ -46,6 +46,13 @@ public class ShareFileStorageResourceOptions
         private Metadata _fileMetadata = default;
         internal bool _isFileMetadataSet = false;
 
+        /// <summary>
+        /// Optional. Specifies whether we should skip validating the protocol of the resource before starting the transfer.
+        /// By default this value is set to false.
+        /// Applies to copy, upload, and download transfers.
+        /// </summary>
```

Comment:
We should add something like "Protocol validation requires share-level read access"